File: solutions/javascript/space-age/1/space-age.js

export const age = (planet, seconds) => {
    let result = seconds / 31557600;
    switch (planet) {
        case 'mercury': result /= 0.2408467; break;
        case 'venus': result /= 0.61519726; break;
        case 'mars': result /= 1.8808158; break;
        case 'jupiter': result /= 11.862615; break;
        case 'saturn': result /= 29.447498; break;
        case 'uranus': result /= 84.016846; break;
        case 'neptune': result /= 164.79132; break;
    }
    return Number(result.toFixed(2));
};

File: solutions/go/space-age/1/space_age.go
package space

type Planet string

func Age(seconds float64, planet Planet) float64 {
	result := seconds / 31557600.0
	switch planet {
	case "Earth":
	case "Mercury":
		result /= 0.2408467
	case "Venus":
		result /= 0.61519726
	case "Mars":
		result /= 1.8808158
	case "Jupiter":
		result /= 11.862615
	case "Saturn":
		result /= 29.447498
	case "Uranus":
		result /= 84.016846
	case "Neptune":
		result /= 164.79132
	default:
		result = -1.0
	}
	return result
}
